README.md

```diff
@@ -1 +1,491 @@
+    self_class = self.class
+    array = self_class == Array ? self : flatten
+    while i < array.length
+      self_class == Array ? yield(array[i]) : yield(array[i], array[i + 1])
+      i += self_class == Array ? 1 : 2
+    end
+  end
+
+  def my_each_with_index
+    # If no block is given, an enumerator is returned instead.
+    return to_enum unless block_given?
+
+    array = self.class == Array ? self : to_a
+    i = 0
+    while i < length
+      yield(array[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    # If no block is given, an Enumerator is returned instead.
+    return to_enum unless block_given?
+
+    enumerable = self.class == Array ? [] : {}
+    if enumerable.class == Array
+      my_each do |n|
+        enumerable.push(n) if yield(n)
+      end
+    else
+      my_each do |key, value|
+        enumerable[key] = value if yield(key, value)
+      end
+    end
+    enumerable
+  end
+
+  def my_all?
+    # Return true if no block is given
+    return true unless block_given?
+
+    boolean = true
+    if self.class == Array
+      my_each do |n|
+        boolean = false unless yield(n)
+      end
+    else
+      my_each do |key, value|
+        boolean = false unless yield(key, value)
+      end
+    end
+    boolean
+  end
+
+  def my_any?
+    # Return true if no block is given
+    return true unless block_given?
+
+    boolean = false
+    if self.class == Array
+      my_each do |n|
+        boolean = true if yield(n)
+      end
+    else
+      my_each do |key, value|
+        boolean = true if yield(key, value)
+      end
+    end
+    boolean
+  end
+
+  def my_none?
+    # Return false in case no block is provided
+    return false unless block_given?
+
+    boolean = true
+    if self.class == Array
+      my_each do |n|
+        boolean = false if yield(n)
+        break unless boolean
+      end
+    else
+      my_each do |key, value|
+        boolean = false if yield(key, value)
+        break unless boolean
+      end
+    end
+    boolean
+  end
+
+  def my_count(element = nil)
+    counter = 0
+    if block_given?
+      if self.class == Array
+        my_each do |n|
+          counter += 1 if yield(n)
+        end
+      else
+        my_each do |key, value|
+          counter += 1 if yield(key, value)
+        end
+      end
+    elsif !block_given? && element.nil?
+      return length
+    elsif !block_given? && !element.nil?
+      my_each do |n|
+        counter += 1 if n == element
+      end
+    end
+    counter
+  end
+
+  def my_map
+    # If no block is given, an enumerator is returned instead.
+    return to_enum unless block_given?
+
+    array = []
+    if self.class == Array
+      my_each do |n|
+        array << yield(n)
+      end
+    else
+      my_each do |key, value|
+        array << yield(key, value)
+      end
+    end
+    array
+  end
+
+  def my_inject(symbol = nil, initial_value = nil)
+    if symbol.class != Symbol
+      temp = symbol
+      symbol = initial_value
```

Comment:
```
# Same using a block and inject
(5..10).inject { |sum, n| sum + n }            #=> 45
# Same using a block
(5..10).inject(1) { |product, n| product * n } #=> 151200
# find the longest word
longest = %w{ cat sheep bear }.inject do |memo, word|
   memo.length > word.length ? memo : word
end
longest                                        #=> "sheep"
```

Reply:
The method was updated to successfully pass all test cases 👍